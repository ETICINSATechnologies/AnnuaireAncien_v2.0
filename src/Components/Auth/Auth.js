class Auth {
    static connect(jwt){
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(){
        sessionStorage.removeItem('token')
    }

    static isConnected(){
        return !!sessionStorage.getItem('token');
    }

    static getToken() {
        return sessionStorage.getItem('token')
    }

    static getPayload(){
        return sessionStorage.getItem('token');
    }

    static addCorrectButton(buttons){
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
