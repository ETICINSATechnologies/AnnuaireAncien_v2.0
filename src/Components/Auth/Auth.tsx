class Auth {
    static connect(jwt: string) {
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(): void {
        sessionStorage.removeItem('token')
    }

    static isConnected(): boolean {
        return !!sessionStorage.getItem('token');
    }

    static getToken(): string {
        return sessionStorage.getItem('token') as string;
    }

    static addCorrectButton(buttons: Array<string>) {
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
