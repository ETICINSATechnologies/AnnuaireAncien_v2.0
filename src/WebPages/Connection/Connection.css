.Connection {
  display: grid;

  width: 100%;
  height: 100%;
}

.Connection .connect_section {
  display: grid;
  height: 100%;
}

.Connection .connect_form {
  display: grid;
  grid-template-rows: 1fr repeat(5, 6.2vmin) 10fr;

  background-color: var(--backgroundContainer);

  justify-self: center;
  vertical-align: middle;
  margin: 10vh auto;
  padding: 1%;

  border: 0.45vmin solid black;

  width: 40vmin;
  height: 50vmin;
}

.Connection .info_area {
  color: red;
  text-align: center;
  margin: auto;
  font-size: 2.4vmin;
  width: 90%;
  height: 90%;
}

.Connection .connect_form p {
  margin: auto auto 0 auto;
  font-size: 3.8vmin;
  width: 95%;
  height: 4vmin;
}

.Connection .connect_form a {
  margin: auto;

  text-decoration: none;
  color: var(--passwordLostConnection);

  font-size: 3vmin;
}
.Connection .connect_form a:hover {
  text-decoration: underline;
}

.Connection .connect_form input[type="text"],
input[type="password"] {
  width: 95%;
  height: 4vmin;
  margin: auto;
  padding-left: 0.7vmin;
  font-size: 3vmin;
  background-color: white;
  justify-self: center;
}

.Connection .connect_input {
  cursor: pointer;
  color: white;
  background: var(--backgroundButton);
  border: 0.3vmin solid black;

  margin: auto;
  font-size: 3.5vmin;
  width: 60%;
  height: 6vmin;
}
