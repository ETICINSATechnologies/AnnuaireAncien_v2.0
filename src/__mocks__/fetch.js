import member from "../Tools/member";
import members from "../Tools/members";
import board from "../Tools/board";

export default (url) => {
    let expectedResponse = null;

    if (/auth\/login$/.test(url))
        expectedResponse = {token: 'this is a test token'};
    else if (/core\/member\/(\d*|me)$/.test(url))
        expectedResponse = member;
    else if (/core\/member\/board\/latest$/.test(url))
        expectedResponse = board;
    else if (/core\/member.*/.test(url))
        expectedResponse = members;
    else
        return null;

    return Promise.resolve({
        status: 200,
        json: () => {
            return Promise.resolve(
                expectedResponse
            )
        }
    })
};
