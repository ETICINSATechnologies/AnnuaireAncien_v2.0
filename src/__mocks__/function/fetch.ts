import member from "../data/member.json";
import members from "../data/members.json";
import board from "../data/board.json";
import departments from "../data/departments.json";
import positions from "../data/positions.json";

export function fetch(url: string, param?: any): Promise<any> {
    let expectedResponse: object;

    if (/auth\/login$/.test(url))
        expectedResponse = {token: 'this is a test token'};
    else if (/core\/member\/(\d*|me)$/.test(url))
        expectedResponse = member;
    else if (/core\/member\/board\/latest$/.test(url))
        expectedResponse = board;
    else if (/core\/member$/.test(url))
        expectedResponse = members;
    else if (/core\/department$/.test(url))
        expectedResponse = departments;
    else if (/core\/position/.test(url))
        expectedResponse = positions;

    return Promise.resolve({
        status: 200,
        json: () => {
            return Promise.resolve(
                expectedResponse
            )
        }
    })
}
