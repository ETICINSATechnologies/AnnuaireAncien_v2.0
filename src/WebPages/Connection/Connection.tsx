import React, {Component} from 'react';
import {Link, Redirect} from 'react-router-dom'
import './Connection.css';
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import Auth from "../../Components/Auth/Auth";


type ConnectionState = {
    parameters: Param
    status: string
}

interface Param {
    username: string
    password: string
}

const statusMessage = {
    connect: '',
    loading: '',
    error: 'Une erreur est survenue',
    invalid: 'Identifiant ou mot de passe invalide',
    missingValue: 'Veuillez compléter tous les champs du formulaire !'
};

class Connection extends Component<{}, ConnectionState> {
    state = {
        parameters: {
            username: "username",
            password: "password"
        },
        status: 'loading',
    };

    onChange = (event: React.ChangeEvent) => {
        let parameters = Object.assign({}, this.state.parameters);
        (parameters as any)[event.target.className] = (event.target as any).value;
        this.setState({parameters: parameters});
    };

    tryToConnect = (event: React.MouseEvent) => {
        event.preventDefault();
        if (!this.state.parameters.username || !this.state.parameters.password) {
            this.setState({status: 'missingValue'})
        }
        else {
            fetch('/login', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify(this.state.parameters)
            })
                .then(res => {
                        if (res.status === 200) {
                        this.setState({status: 'pending'});
                        res.json()
                            .then(result => {
                                if(this.state.parameters.username ==='admin'){
                                    Auth.connect(result.token, true);
                                }else {
                                    Auth.connect(result.token, false);
                                }
                                this.setState({status: 'connect'})
                            })
                    }
                    else if (res.status === 401)
                        this.setState({status: 'invalid'});
                    else
                        this.setState({status: 'error'});
                })
        }
    };

    render() {
        let activeButton = ["home"];

        return (
            <React.Fragment>
                <Header/>
                <Nav buttons={activeButton}> </Nav>
                <div className="Connection">
                    <section className="connect_section">
                        <form className="connect_form">
                            <div className="info_area"> {(statusMessage as any)[this.state.status]} </div>
                            <p> Identifiant </p>
                            <input type="text" className="username" value={this.state.parameters.username}
                                   onChange={this.onChange}/>
                            <p> Mot de passe </p>
                            <input type="password" className="password" value={this.state.parameters.password}
                                   onChange={this.onChange}/>
                            <Link to="/recovery"> Mot de passe oublié ?</Link>
                            <input type="submit" className="connect_input" value="Se connecter"
                                   onClick={this.tryToConnect}/>
                        </form>
                    </section>
                    {this.state.status === 'connect' && Auth.isAdmin() && <Redirect to='/admin'/>}
                    {this.state.status === 'connect' && !Auth.isAdmin() && <Redirect to='/profile'/>}
                </div>
            </React.Fragment>
        );
    }
}

export default Connection;
