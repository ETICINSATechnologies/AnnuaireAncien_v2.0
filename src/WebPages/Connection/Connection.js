import React, {Component} from 'react';
import {Redirect} from 'react-router-dom'
import './Connection.css';
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import Auth from "../../Components/Auth/Auth";
import fetch from "../../__mocks__/fetch";

class Connection extends Component {
    constructor(props) {
        super(props);
        this.state = {
            parameters: {
                username: "username",
                password: "password"
            },
            status: 'loading',
            statusMessage: {
                connect: '',
                loading: '',
                invalid: 'Identifiant ou mot de passe invalide',
                missingValue: 'Veuillez compléter tous les champs du formulaire !'
            }
        };

        this.onChange = this.onChange.bind(this);
        this.tryToConnect = this.tryToConnect.bind(this);
    }

    onChange(event) {
        let parameters = Object.assign({}, this.state.parameters);
        parameters[event.target.className] = event.target.value;
        this.setState({parameters: parameters});
    }

    tryToConnect(event) {
        event.preventDefault();
        if (!this.state.parameters.username || !this.state.parameters.password) {
            this.setState({status: 'missingValue'})
        }
        else {
            fetch('api/v1/auth/login', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify(this.state.parameters)
            })
                .then(res => {
                    if (res.status !== 200)
                        this.setState({status: 'invalid'});
                    else {
                        this.setState({status: 'pending'});
                        res.json()
                            .then(result => {
                                console.log("token: " + result.token);
                                Auth.connect(result.token);
                                this.setState({status: 'connect'})
                            })
                    }
                })
        }
    }

    render() {
        let activeButton = ["home"];

        return (
            <React.Fragment>
                <Header/>
                <Nav buttons={activeButton}> </Nav>
                <div className="Connection">
                    <section className="connect_section">
                        <form className="connect_form">
                            <div className="info_area"> {this.state.statusMessage[this.state.status]} </div>
                            <p> Identifiant </p>
                            <input type="text" className="username" value={this.state.parameters.username}
                                   onChange={this.onChange}/>
                            <p> Mot de passe </p>
                            <input type="text" className="password" value={this.state.parameters.password}
                                   onChange={this.onChange}/>
                            <a href="/recup"> Mot de passe oublié ?</a>
                            <input type="submit" className="connect_input" value="Se connecter"
                                   onClick={this.tryToConnect}/>
                        </form>
                    </section>
                    {this.state.status === 'connect' && <Redirect to='/profile'/>}
                </div>
            </React.Fragment>
        );
    }
}

export default Connection;
