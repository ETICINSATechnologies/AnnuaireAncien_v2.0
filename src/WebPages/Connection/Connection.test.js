import React from 'react';

import Connection from './Connection';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Auth from '../../Components/Auth/Auth';
import fetch from '../../__mocks__/fetch'
Enzyme.configure({adapter: new Adapter()});

describe( 'Connection component' , () => {

    it('renders without crashing', () => {
        const tree = renderer.create(<Connection />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Connection unit test', () => {
        expect(
            Enzyme.shallow(<Connection/>)
                .find('div.Connection')
                .length
        ).toBe(1)
    });

    it('Connection button clicked', () => {
        // fetch.url = jest.fn( () => 'api/v1/auth/login' );
        const wrapper = Enzyme.shallow(<Connection/>);
        wrapper.find(".connect_input").simulate('click', { preventDefault() {} });
        expect(wrapper.state("status")).toBeTruthy();
        // expect(wrapper.state("status")).equals("connect");
    });



});