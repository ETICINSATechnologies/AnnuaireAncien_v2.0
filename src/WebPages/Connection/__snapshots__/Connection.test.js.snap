// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection component renders without crashing 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/"
      >
        accueil
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <div
    className="Connection"
  >
    <section
      className="connect_section"
    >
      <form
        className="connect_form"
      >
        <div
          className="info_area"
        >
           
          
           
        </div>
        <p>
           Identifiant 
        </p>
        <input
          className="username"
          onChange={[Function]}
          type="text"
          value="username"
        />
        <p>
           Mot de passe 
        </p>
        <input
          className="password"
          onChange={[Function]}
          type="text"
          value="password"
        />
        <a
          href="/recup"
        >
           Mot de passe oubli√© ?
        </a>
        <input
          className="connect_input"
          onClick={[Function]}
          type="submit"
          value="Se connecter"
        />
      </form>
    </section>
  </div>,
]
`;
